### ISSUE TITLE
Unexpected characters rendered when format- or label- ends with '}'

### ISSUE THREAD

**Issue by ewen-lbh** [polybar/polybar#2331]
## Checklist

* [x] I have read the appropriate section in the [contributing
  guidelines](https://github.com/polybar/polybar/blob/master/CONTRIBUTING.md)
* [x] I believe this issue is a problem with polybar itself and not a misconfiguration on my part.
* [x] I have searched for other open and closed [issues](https://github.com/polybar/polybar/issues?q=is%3Aissue) that
  may have already reported this problem.
* [x] I have checked the [known issues](https://github.com/polybar/polybar/wiki/Known-Issues) page for this problem.

## Describe the bug

**Expected behavior:**
A `}` at the end of a `format-*` or a `label-*` does not cause the result to be prefixed with unexpected characters, and is rendered as a single `}`.

**Actual behavior:**
A closing curly bracket U+007D `}` is rendered as `PR}` when placed at the end and used in the `format-*` or `label-*` properties from the `internal/battery` or `internal/network` modules.

When used in the `internal/network` module, the bracket is replaced by an even weirder sequence: `A A A PR}`.

Similarly, the bracket gets replaced with `A PR}` when used in `internal/date`.

This does not happen when the brace is placed anywhere else in the string. Ending the format/label string with an _opening_ brace has the normal and expected behavior too.

**Was it working before?**
* Did you also experience this bug in an earlier version of polybar **no**
* If no, what was the last version where this worked correctly?
`3.4.` (I don't remember the patch version number, sorry.)

## To Reproduce

```dosini
[section/bar-base]
background = #00000000
height=50

[bar/top]
inherit = section/bar-base
modules-right = battery
; tray-position = left

[module/battery]
type = internal/battery

battery = BAT0
adapter = AC0

format-charging = <label-charging>
format-discharging = <label-discharging>
label-full = [FULL}
label-charging = [%percentage:3%}+
label-discharging =[%percentage:3%}
```

## Polybar Log

```sh-session
$ polybar --stdout --config=config.ini --log=trace top
notice: Parsing config file: config.ini
- config_parser: Parsing config.ini
- config: Inheriting keys from "section/bar-base" in "bar/top"
* Loaded monitor eDP1 (1920x1080+0+0)
* Bar geometry: 1920x50+0+0; Borders: 0,0,0,0
- bar: Attach X event sink
- bar: Attach signal receiver
- controller: Install signal handler
- controller: Setup user-defined modules
- module/battery: Attach inotify at /sys/class/power_supply/BAT0/energy_now
- module/battery: Attach inotify at /sys/class/power_supply/AC0/online
* Starting application
- controller: Main thread id = 1
* Starting module/battery
* Entering event loop (thread-id=1)
- module/battery: Thread id = 2
* Eventqueue worker (thread-id=3)
* module/battery: Rebuilding cache
%{r}[%percentage% PR}
* module/battery: Rebuilding cache
%{r}[ 11 PR}
* module/battery: Polling values (inotify fallback)
- module/battery: Inotify event reported for /sys/class/power_supply/AC0/online
- module/battery: Inotify event reported for /sys/class/power_supply/BAT0/energy_now
* module/battery: Polling values (inotify fallback)
- module/battery: Inotify event reported for /sys/class/power_supply/AC0/online
- module/battery: Inotify event reported for /sys/class/power_supply/BAT0/energy_now
* module/battery: Rebuilding cache
%{r}[ 11 PR}
```

## Screenshots

![2020-12-24-222302_109x42_scrot](https://user-images.githubusercontent.com/39094199/103106536-a0823c00-4636-11eb-9e8c-5d8b5c77c9e7.png)

## Environment:
* WM: i3-gaps
* Distro: Manjaro Linux x86_64
* Output of `polybar -vvv`:

```
polybar 3.5.3

Features: +alsa +curl +i3 +mpd +network(libnl) +pulseaudio +xkeyboard

X extensions: +randr (+monitors) +composite +xkb +xrm +xcursor

Build type: Release
Compiler: /usr/bin/c++
Compiler flags: -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -Wall -Wextra -Wpedantic -O3 -DNDEBUG
Linker flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now 
```

## Additional context

I installed polybar 3.5 via `aur/polybar` using [yay](https://aur.archlinux.org/packages/yay/)

**Comment by patrick96** [comment]
Thanks for the detailed report. Polybar has an "optimization" where it just replaces all occurences of `}%{` with a space. I knew that this would come back to bite us at some point.

**Comment by patrick96** [comment]
Fixed in #2332
